BoneReconstructionPlanner/BoneReconstructionPlanner.py

```diff
@@ -809,13 +823,16 @@ def generateFibulaPlanes(self):
           intersectionsList[j-1].GetRASBounds(bounds0)
           intersectionsList[j].GetRASBounds(bounds1)
 
+        #calculate how much each FibulaPlaneA should be translated so that it doesn't intersect with fibulaPlaneB
         z0Sup = bounds0[5]
+        z0Inf = bounds0[4]
+        z1Sup = bounds1[5]
         z1Inf = bounds1[4]
-        deltaz = z0Sup - z1Inf
+        deltaZ = (z0Sup - z0Inf)/2 + (z1Sup - z1Inf)/2
 
-        betweenSpace.append(deltaz)
+        betweenSpace.append(deltaZ)
 
-        self.fibula2FibulaPlanesPositionA.append(self.fibula2FibulaPlanesPositionB[i-1] + fibulaZ*(betweenSpace[i-1]+additionalBetweenSpaceOfFibulaPlanes))
+        self.fibula2FibulaPlanesPositionA.append(self.fibula2FibulaPlanesPositionB[i-1] + fibulaZ*(intersectionDistanceMultiplier*betweenSpace[i-1]+additionalBetweenSpaceOfFibulaPlanes))
 
       self.fibula2FibulaPlanesPositionB.append(self.fibula2FibulaPlanesPositionA[i] + boneSegmentsDistance[i]*fibulaZ)
 

```

Comment:
Till here. I think this code is hard to understand. Do you have any suggestions on how to improve it?

Reply:
I think this is quite clear. However, I would recommend to split up this very long method to smaller methods and use the method documentation to explain in 1-2 sentences what the module does.